mailpoet/tests/integration/Subscription/ManageTest.php

```diff
@@ -64,59 +79,65 @@ public function testItDoesntRemoveHiddenSegmentsAndCanResubscribe() {
       $this->settings,
       $this->diContainer->get(NewSubscriberNotificationMailer::class),
       $this->diContainer->get(WelcomeScheduler::class),
+      $this->diContainer->get(CustomFieldsRepository::class),
       $this->diContainer->get(SegmentsRepository::class),
       $this->diContainer->get(SubscribersRepository::class),
-      $this->diContainer->get(SubscriberSegmentRepository::class)
+      $this->diContainer->get(SubscriberSegmentRepository::class),
+      $this->diContainer->get(SubscriberSaveController::class)
     );
     $_POST['action'] = 'mailpoet_subscription_update';
     $_POST['token'] = 'token';
     $_POST['data'] = [
       'first_name' => 'John',
       'last_name' => 'John',
       'email' => 'john.doe@example.com',
-      'status' => Subscriber::STATUS_SUBSCRIBED,
-      'segments' => [$this->segmentB->id],
+      'status' => SubscriberEntity::STATUS_SUBSCRIBED,
+      'segments' => [$this->segmentB->getId()],
     ];
 
     $manage->onSave();
 
-    $subscriber = Subscriber::findOne($this->subscriber->id);
-    $subscriber->withSubscriptions();
-    $subscriptions = array_map(function($s) {
-      return ['status' => $s['status'], 'segment_id' => $s['segment_id']];
-    }, $subscriber->subscriptions);
-    usort($subscriptions, function($a, $b) {
-      return $a['segment_id'] - $b['segment_id'];
-    });
-    expect($subscriber->status)->equals(Subscriber::STATUS_SUBSCRIBED);
+    $subscriber = $this->subscribersRepository->findOneById($this->subscriber->getId());
+    if (!$subscriber) throw new InvalidStateException();
```

Comment:
It feels a little strange to use an exception like this in a test. I'm assuming this is for PHPStan?

It would feel a little better to me to use something like `assert($subscriber instanceof SubscriberEntity);` which I think PHPStan understands, but it's not a big deal one way or another.

Reply:
If I am not wrong, we want to avoid using the `assert` function. We have [a ticket](https://mailpoet.atlassian.net/browse/MAILPOET-4258) for that. But you are right that the exception looks a bit weird here, so I would change it to `$this->assertInstanceOf(SubscriberEntity::class, $subscriber);`. Do you agree with it?

Reply:
@lysyjan I was familiar with wanting to remove `assert` from the codebase but I didn't realize that applied to our test code as well. Thanks for pointing that out!

The reason I suggested `assert` is because I tried `expect($subscriber)->isInstanceOf(SubscriberEntity::class);`, which PHPStan didn't seem to understand. If PHPStan does understand `$this->assertInstanceOf(SubscriberEntity::class, $subscriber);` I would definitely prefer that!